datacube/index/memory/_products.py

```diff
@@ -146,7 +146,7 @@ def get_by_name_unsafe(self, name: str) -> Product:
         return self.clone(self.by_name[name])
 
     @override
-    def search_robust(self, **query: QueryField) -> Iterable[tuple[Product, QueryDict]]:
+    def search_robust(self, **query) -> Iterable[tuple[Product, QueryDict]]:
```

Comment:
Dropped the `QueryField` here since the memory index driver implementation gave a type error when adding that. Someone who knows this stuff better should look into that.